containers/conformance.md

```diff
@@ -0,0 +1,220 @@
+- **Source Build**: must be string `-sources` appended to the end of tag.
+  * If this is a source build, the tag must contain full version number (major+minor+patch) information.
+  * Linux Distro information is recommended.
+  * Must NOT contain customize build information.
+  * For example: `1.23.0-ubi-sources`.
+
+For example, these are valid image tags:
+
+- latest
+- latest-ubuntu
+- latest-ubuntu-sources
+- latest-ubi
+- latest-ubi-sources
+- lts
+- lts-ubuntu
+- lts-ubi
+- 1
+- 1-ubuntu
+- 1-ubi
+- 1.23
+- 1.23-ubuntu
+- 1.23-ubi
+- 1.23.0
+- 1.23.0-ubuntu
+- 1.23.0-ubuntu-sources
+- 1.23.0-ubi
+- 1.23.0-ubi-sources
+- 1.23.0-alpine.pr-1234
+- 1.23.0-ubi.users-john-test1
+
+### Files and Directories
+
+This is required folder structure for all Zowe components:
+
+```
+/licenses
+/app
+  +- README.md
+```
+
+- `/licenses` folder holds all license related files. It should include at least the license information for current application. It's recommended to include license notice file of all pedigree dependencies. All licenses files must be in UTF-8 encoding.
+- `/app/README.md` provides information about the application for end-user.
+
+These file(s) and folder(s) are recommended:
+
+```
+/app
+  +- manifest.json or manifest.yaml
+  +- /bin/<lifecycle-scripts>
+  +- <other-application-files>
+```
+
+- `/app/manifest.(json|yaml)` is recommended for Zowe component. The format of the file is defined at [Zowe component manifest](https://docs.zowe.org/stable/extend/packaging-zos-extensions/#zowe-component-manifest). Component must use same manifest file when it's running on z/OS.
+- `/app/bin/<lifecycle-scripts>` should remain same as what it is running on z/OS. Component must use same lifecycle scripts when it's running on z/OS.
+
+### Environment Variable(s)
+
+These environment variable(s) must be set as a fixed value in the image:
+
+- `ZOWE_COMPONENT_ID`: this is the Zowe component ID for current image. For example: `ENV ZOWE_COMPONENT_ID=gateway`.
+
+### User `zowe`
+
+In the Dockerfile, a `zowe` user and group must be created. Example command `RUN groupadd -r zowe && useradd --no-log-init -r -g zowe zowe`.
+
+`USER zowe` must be specified before the first `CMD` or `ENTRYPOINT`.
+
+### Multi-Stage Build
+
+Multi-stage build is recommended to keep image small and concise. Learn more from [Use multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/).
+
+## Runtime
+
+This section is mainly for information. No actions required for components except where it's specified explicitly.
+
+Below sections are mainly targeting Kubernetes or OpenShift environment. Starting Zowe containers in Docker environment with `docker-compose` is in plan and may change some of the requirements.
+
+### General rules
+
+**Components MUST:**
+
+- NOT be started as root user in the container.
+- listen on ONLY one port in the container.
+- NOT rely on hardcoded directory names like `/app`.
```

Comment:
above it says /app is required

Reply:
Yes /app is required but the application should not assume the runtime is /app. That's what it means here. I will try to rephrase it.